File: keejob store-angular/src/app/models/avis.ts
import { CentreFormation } from "./centre-formation";
import { Certification } from "./certification";
import { Evaluation } from "./evaluation";
import { FormationKeejob } from "./formation-keejob";

export interface Avis {
  id?: number;
  nomAuteur: string;
  poste: string;
  note: number;          // 1 à 5
  commentaire: string;
  date: string;           // format ISO "yyyy-MM-dd" (LocalDate côté Java)
  centre?: CentreFormation | { id: number }; // selon besoin d'affichage ou juste la ref
  formation?: FormationKeejob | { id: number };
  evaluation?: Evaluation | { id: number };
  certification?: Certification | { id: number };

}

File: keejobStore/src/main/java/com/example/keejobstore/entity/Avis.java
package com.example.keejobstore.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.time.LocalDate;

@Setter
@FieldDefaults(level = AccessLevel.PRIVATE)
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Avis {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nomAuteur;   // "Yassine K."
    private String poste;       // "Développeur Web"
    private Integer note;       // 1 à 5

    @Column(length = 1000)
    private String commentaire;

    private LocalDate date;

    @ManyToOne
    @JoinColumn(name = "centre_id")
    @JsonIgnore
    private CentreFormation centre;
}